File: app/src/main/java/com/sabanciuniv/todoapp/model/ToDo.java
package com.sabanciuniv.todoapp.model;

import java.io.Serializable;
import java.time.LocalDateTime;

public class ToDo implements Serializable {
    private String toDo, title;
    private String dueDate, id;
    private boolean isChecked, isLoading;

    public ToDo(String toDo, String title, String dueDate, boolean isChecked, String id) {
        this.toDo = toDo;
        this.title = title;
        this.dueDate = dueDate;
        this.isChecked = isChecked;
        this.id = id;
        this.isLoading = false;
    }

    public ToDo() {}

    public String getToDo() {
        return toDo;
    }

    public void setToDo(String toDo) {
        this.toDo = toDo;
    }

    public String getDueDate() {
        return dueDate;
    }

    public void setDueDate(String dueDate) {
        this.dueDate = dueDate;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isChecked() {
        return isChecked;
    }

    public void setChecked(boolean checked) {
        isChecked = checked;
    }

    public String getId() {
        return id;
    }

    public boolean isLoading() {
        return isLoading;
    }

    public void setLoading(boolean loading) {
        isLoading = loading;
    }
}

File: app/src/main/java/com/sabanciuniv/todoapp/model/ToDo.kt
package com.sabanciuniv.todoapp.model

import org.json.JSONObject
import java.io.Serializable
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class ToDo : Serializable {
    var toDo: String? = null
    var title: String? = null
    var dueDate: LocalDateTime? = null
    var id: String? = null
        private set
    var isChecked = false
    var isLoading = false

    constructor(toDo: String?, title: String?, dueDate: LocalDateTime?, isChecked: Boolean, id: String?) {
        this.toDo = toDo
        this.title = title
        this.dueDate = dueDate
        this.isChecked = isChecked
        this.id = id
        isLoading = false
    }

    constructor()
    fun getShortened(): String{

        if(this.title!!.length > 25){
            return this.title!!.substring(0,  25) + "..."
        }
            return this.title!!
    }

    companion object {
        // Define a DateTimeFormatter for parsing the date and time strings
        private val dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")

        fun fromBackendJson(currentToDo: JSONObject): ToDo {
            return ToDo(
                currentToDo.getString("description"),
                currentToDo.getString("title"),
                LocalDateTime.parse(currentToDo.getString("dueDate"), dateFormatter),
                currentToDo.getBoolean("checked"),
                currentToDo.getString("id")
            )
        }
    }
}